### PR TITLE
feat: gradio app

### PR DESCRIPTION
**Pull request by sebastian-weisshaar**
Gradio app to host an inference model. Lora weights are downloaded on instantiation of the model.

### CODE REVIEW

**Review by samsja — CHANGES_REQUESTED** [review]
I think that you need to remove this artifacts folder

File: gradio/app.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+MODEL_CONFIG = AutoConfig.from_pretrained(BASE_MODEL, trust_remote_code=True)
+PEFT_CONFIG = PeftConfig.from_pretrained(
+    LORA_WEIGHTS,
+    trust_remote=True,
+)
+model = AutoModelForCausalLM.from_pretrained(
+    pretrained_model_name_or_path=BASE_MODEL,
+    torch_dtype=torch.float16,
+    device_map='auto',
+    config=MODEL_CONFIG,
```
Comment (samsja) [inline gradio/app.py:30]: I would expect that this code is the same as the one used in the run code eval, is there a way we can refactor to only have this code once?

Comment (samsja) [reply gradio/app.py:30]: Ideally, we would even have a small cli to select which model to run

Comment (sebastian-weisshaar) [reply gradio/app.py]: Both done

**Pull request merged**